### PR TITLE
doc: rewrite job-control example

### PR DESCRIPTION
**Pull request by mhinz**
The previous example was outdated, so I took the chance to make the example a bit more interesting.

Fixes https://github.com/neovim/neovim/issues/7009

### CODE REVIEW

File: runtime/doc/job_control.txt [hunk 1]
```diff
@@ -102,36 +102,29 @@ function. Here's a more object-oriented version of the above:
...
+      let object.id = jobstart(object.cmd, object)
+      return object
     endfunction
 
-    let s1 = Shell.new('1')
-    let s2 = Shell.new('2', 'for i in {1..10}; do echo hello $i!; sleep 1; done')
-
-
+    let instance = Shell.new('bomb',
+          \ 'for i in {9..1}; do echo $i 1>&$((i % 2 + 1)); sleep 1; done')
```
Comment (justinmk) [inline runtime/doc/job_control.txt]: `1>&` is a bashism, would be better to keep it shell agnostic (POSIX).

File: runtime/doc/job_control.txt [hunk 2]
```diff
@@ -102,36 +102,29 @@ function. Here's a more object-oriented version of the above:
...
-      let instance = extend(copy(g:Shell), {'name': a:name})
-      let argv = ['bash']
-      if a:0 > 0
-        let argv += ['-c', a:1]
-      endif
-      let instance.id = jobstart(argv, instance)
-      return instance
+    function Shell.new(name, cmd)
+      let object = extend(copy(g:Shell), {'name': a:name})
+      let object.cmd = ['bash', '-c', a:cmd]
```
Comment (justinmk) [inline runtime/doc/job_control.txt]: `'sh'`

File: runtime/doc/job_control.txt (hunk 1 above)
Comment (justinmk) [inline runtime/doc/job_control.txt]: POSIX version:

    sh -c 'for i in $(seq 9 -1 1); do echo $i 1>&$((i % 2 + 1)); sleep 1; done'

File: runtime/doc/job_control.txt [hunk 3]
```diff
@@ -102,36 +102,29 @@ function. Here's a more object-oriented version of the above:
 >
     let Shell = {}
 
-    function Shell.on_stdout(job_id, data) dict
-      call append(line('$'), self.get_name().' stdout: '.join(a:data))
+    function Shell.on_stdout(_job_id, data, event)
+      call append(line('$'),
```
Comment (justinmk) [inline runtime/doc/job_control.txt]: `normal! G` before this line will ensure the user sees the activity.

Comment (mhinz) [inline runtime/doc/job_control.txt]: I kind of liked that the user is free to move around while this is happening, but you're right that we should jump to the end at least once.

How about..

```vim
function Shell.new(name, cmd)
  let object = extend(copy(g:Shell), {'name': a:name})
  let object.cmd = ['sh', '-c', a:cmd]
  let object.id = jobstart(object.cmd, object)
  $
  return object
endfunction
```

Comment (justinmk) [inline runtime/doc/job_control.txt]: ok

**Pull request merged**